CMakeLists.txt

```diff
@@ -4,11 +4,27 @@ ELSE()
   SET(KOKKOS_HAS_TRILINOS OFF CACHE BOOL "")
 ENDIF()
 
+set(Kokkos_SOURCE_DIR ${CMAKE_SOURCE_DIR})
+
 IF(NOT KOKKOS_HAS_TRILINOS)
   cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
   project(Kokkos CXX)
 
-  INCLUDE(cmake/kokkos.cmake)
```

Comment:
Does this PR preserve the behavior of this "pure CMake" build system that @bjoo and others rely on?

Reply:
In theory yes, in practice, it needs more testing to ensure that it is working.  The last commit shows the changes.